discretisedfield/operators.py

```diff
@@ -8,3 +11,92 @@ def integrate(field, direction=None, cumulative=False):
 
     """
     return field.integrate(direction=direction, cumulative=cumulative)
+
+
+def _1d_diff(order, array, dx):
+    """Differentiate 1D array."""
+
+    # Directional derivative cannot be computed if less or an equal number of
+    # discretisation cells exists in a specified direction than the order.
+    # In that case, a zero array is returned.
+    if len(array) < order + 1:
+        return np.zeros_like(array)
+
+    if order == 1:
+        # Second order accuracy is in the center of the array and
+        # first order at the boundaries.
+        derivative_array = np.gradient(array, dx, edge_order=1)
+        if len(array) > 3:
```

Comment:
```suggestion
        if len(array) < 3:
            # Second order accuracy is in the center of the array and
            # first order at the boundaries.
            derivative_array = np.gradient(array, dx, edge_order=1)
        else:
```

We can use `edge_order=2` already for 3 elements.